duet/futuretools.py

```diff
@@ -67,6 +70,9 @@ def cancel(awaitable_future: Future):
                 future.cancel()
 
         def callback(future: FutureLike[T]):
+            if future.cancelled():
+                return
+
```

Comment:
I'd suggest that we ensure the awaitable future gets cancelled here, just in case someone decides to interact with the wrapped future directly:
```python
if future.cancelled():
    awaitable.cancel()
else:
    error = future.exception()
    if error is None:
        awaitable.try_set_result(future.result())
    else:
        awaitable.try_set_exception(error)
```